### PR TITLE
Remove JSON fixture comparison tests, refactor date comparison tests.

### PR DESCRIPTION
**Pull request by gnott**
As a result of reviewing PR https://github.com/elifesciences/elife-article/pull/41, thanks @seanwiseman for the first attempt at adding Python 3 support there.

I reviewed the tests, and I made some change which are hopefully a good addition
* Removed the JSON fixture comparison - these were initially just a quick way to compare the attributes of the article object, and later the values were compared on an attribute by attribute basis, making the headache of the JSON fixtures no longer required (I think)
* Changed the date comparison tests to not need to compare a single string value

I left in the article object's ``pretty()`` method in case it might be useful in the future, but the tests do not depend on it anymore.

### CODE REVIEW

File: tests/test_article.py [hunk 1]
```diff
@@ -196,21 +196,6 @@ def test_get_funder_identifier_fail(self):
         self.funding_award.institution_id = institution_id
         self.assertEqual(self.funding_award.get_funder_identifier(), None)
 
-        # Original:
-        self.assertRaises(ValueError, self.funding_award.get_funder_identifier())
```
Comment (seanwiseman) [inline tests/test_article.py]: I removed this as I do not believe you need it as the above assertion handles the failure case of `get_funder_identifier`. (remove comments below detail the cases)

**Comment by seanwiseman** [comment]
I seem to be getting intermittent failures on the below assertions when run under Python 3 (Python 2 does not seem to fail on these):

```
test_parse_deep.py: line 130

self.assertEqual(len(article_object.datasets[0].authors), 2)
Expected :1
Actual   :2
```


```
test_parse_deep.py: line 52

self.assertEqual(article_object.datasets[0].accession_id, 'EGAS00001000968')
Expected :None
Actual   :EGAS00001000968
```
It's possible that this is due to some state being heald between tests or a possible dict -> str comparrison order issue. Also, they seem to always succeed when run in isolation, it's just when the whole suite is run you sometimes see these two fail.

File: tests/test_article.py (hunk 1 above)
Comment (gnott) [reply tests/test_article.py]: This was probably just a test in order to get absolutely 100% coverage. That's not realistic so the test can be omitted.

**Comment by gnott** [comment]
I also experienced the same two test failures at times, the author length of 1 instead of 2, and the missing accession_id, when running the tox tests. Running py35 tests on their own it was fine (I think). The tests seem to pass here on travis-ci and Alfred, so I figured it might just be my environment, or tox. I thought it could be the parser issue with not opening files in read binary 'rb' mode, which I may have hacked to test on my system, I cannot remember. It is interesting you also got the same errors.

**Comment by seanwiseman** [comment]
I have been able to get it to fail pretty consistently (about 1 in 5 on average) just using py35. Still can't seem to get it to fail on py27 so I think py27 is fine. If its any further help the `author` list length values are:

When passing: `[u'M Harrison', u'York University']`
When failing: `[u'M Harrison']`

Which when passing, presumably comes from:
```
elife-article/tests/test_data/elife-00666.xml: line:2086

<collab collab-type="author">York University</collab>
```
Any idea why this is sometimes parsed and included and sometimes not?

**Comment by gnott** [comment]
I was reproducing the same errors. Took a couple tries and trying to figure out the reason, and I may have found it: 

Because there are two types of datasets, "generated" and "used", the map of these values was a dict. The result being the datasets array of the Article object didn't always have the same order, and in the 00666 article example, the first "generated" dataset has 2 authors, but the first "used" dataset has 1 author.

In the latest commit the map of the dataset types is converted to an OrderedDict, so the types should be extracted in the same order each time.

**Comment by seanwiseman** [comment]
Will merge this into the `feature-python3-support` branch and complete the last few changes.

**Pull request merged**